### PR TITLE
Ballon Alerts and Baloon Chat update

### PR DESCRIPTION
**Pull request by FluffyGhoster**
Refactored floating messages.
Images that carry the floating messages for the chat are now sized according to the message size and generated per-client.
Refactored overlays management through the procs, ported from TG.
Updated pickup animations to use the new system, minor refactor of it.
Flick overlay visual now uses a /atom/movable/flick_visual carrying object.
Ported various fonts and the /datum/font system.
Floating messages for the chat and actions now use a pixel-perfect font with selected sizes.
Reduced the number of spawn()s required to send messages.

Example of the new font, for normal talking:
image

And for screaming:
image

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
!review

File: interface/skin.dmf [hunk 1]
```diff
@@ -970,18 +970,20 @@ window "mapwindow"
...
-		font-family = "Arial Rounded MT Bold,Arial Black,Arial,sans-serif"
-		font-size = 7
+		font-family = "Grand9K Pixel"
+		font-size = 7pt
 		is-default = true
 		saved-params = "icon-size"
 		on-show = ".winset\"mainwindow.mainvsplit.left=mapwindow\""
 		on-hide = ".winset\"mainwindow.mainvsplit.left=\""
+		style = ".center { text-align: center; } .maptext { font-family: 'Grand9K Pixel'; font-size: 6pt; -dm-text-outline: 1px black; color: white; line-height: 1.0; } .command_headset { font-weight: bold; } .context { font-family: 'Pixellari'; font-size: 12pt; -dm-text-outline: 1px black; }  .subcontext { font-family: 'TinyUnicode'; font-size: 12pt; line-height: 0.75; } .small { font-family: 'Spess Font'; font-size: 6pt; line-height: 1.4; } .big { font-family: 'Pixellari'; font-size: 12pt; } .reallybig { font-size: 12pt; } .extremelybig { font-size: 12pt; } .greentext { color: #00FF00; font-size: 6pt; } .redtext { color: #FF0000; font-size: 6pt; } .clown { color: #FF69BF; font-weight: bold; } .his_grace { color: #15D512; } .hypnophrase { color: #0d0d0d; font-weight: bold; } .yell { font-weight: bold; } .italics { font-family: 'Spess Font'; font-size: 6pt; line-height: 1.4; }"
```
Comment (SleepyGemmy) [inline interface/skin.dmf:986]:
```suggestion
		style = ".center { text-align: center; } .maptext { font-family: 'Grand9K Pixel'; font-size: 6pt; -dm-text-outline: 1px black; color: white; line-height: 1.0; } .command_headset { font-weight: bold; } .context { font-family: 'Pixellari'; font-size: 12pt; -dm-text-outline: 1px black; }  .subcontext { font-family: 'TinyUnicode'; font-size: 12pt; line-height: 0.75; } .small { font-family: 'Spess Font'; font-size: 6pt; line-height: 1.4; } .big { font-family: 'Pixellari'; font-size: 12pt; } .reallybig { font-size: 12pt; } .extremelybig { font-size: 12pt; } .greentext { color: #00FF00; font-size: 6pt; } .redtext { color: #FF0000; font-size: 6pt; } .yell { font-weight: bold; } .italics { font-family: 'Spess Font'; font-size: 6pt; line-height: 1.4; }"
```
leftovers from TG port? not sure if we use `.greentext` or `.redtext` either.

**Review by SleepyGemmy — COMMENTED** [review]
what is the licensing for the fonts (other than Grand9K Pixel) you are adding?

**Review by alsoandanswer — APPROVED** [review]

File: interface/skin.dmf (hunk 1 above)
Comment (FluffyGhoster) [reply interface/skin.dmf:986]: We don't, but there's no reason to not have it there ready

**Comment by Greenjoe12345** [comment]
The way the new style of floating text moves is not nice. Multiple people have said they dislike it.

**Comment by FluffyGhoster** [comment]
> The way the new style of floating text moves is not nice. Multiple people have said they dislike it.

Yea I have seen it, the fun thing is that it's not supposed to move that much up, and it does not do it locally, so we will have to figure out why only on live it decides to go in the stratosphere

File: code/controllers/subsystems/runechat.dm [hunk 2]
```diff
@@ -0,0 +1,14 @@
+TIMER_SUBSYSTEM_DEF(runechat)
+	name = "Runechat"
+	priority = FIRE_PRIORITY_RUNECHAT
+
+	var/list/datum/callback/message_queue = list()
+
+/datum/controller/subsystem/timer/runechat/fire(resumed)
+	. = ..() //poggers
```
Comment (NonQueueingMatt) [inline code/controllers/subsystems/runechat.dm:8]: interesting comment lol

Comment (FluffyGhoster) [reply code/controllers/subsystems/runechat.dm:8]: For what it looks like, this allows to immediately expire all the timers before this run, i am not entirely sure why they went with making the timer subsystem itself tick before elaborating the runechat callbacks to the second function, but it shouldn't hurt as the timers only run when the waiting for them expires, so ¯\_(ツ)_/¯

File: interface/fonts/license.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+Pixellari created by Zacchary Dempsey-Plante. Website indicates free for commercial use.
+(https://www.dafont.com/pixellari.font?fpp=200)
+
+Spess Font created by MTandi (discord) for /tg/station.
+
+Tiny Unicode created by Jakob Riedle/DuffsDevice. Website indicates free for commercial use.
+(https://fontmeme.com/fonts/tiny-unicode-font/)
+
+VCR OSD Mono created by Riciery Leal/mrmanet. Website indicates 100% free, author confirms it's free for all to use.
+(https://www.dafont.com/font-comment.php?file=vcr_osd_mono)
```
Comment (SleepyGemmy) [inline interface/fonts/license.txt:13]: i don't see this font added in this PR.

File: interface/skin.dmf (hunk 1 above)
Comment (SleepyGemmy) [reply interface/skin.dmf:986]: you should preferably remove them if they're not used.

File: interface/fonts/license.txt (hunk 3 above)
Comment (FluffyGhoster) [reply interface/fonts/license.txt:13]: Oh right I didn't put that in, that's used for the ingame screens and it's a whole other level of pain I would rather spare at the moment,

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**